---
title: 'React error boundaries'
category:
  - react
excerpt: "React error boundaries allow you to prevent your React application from completely crashing in the event of an error in your code.\_"
updated_by: 197c1509-8dff-4d72-9898-334084519619
updated_at: 1579055650
id: 23625f8f-415a-470c-b61e-66c2e0742f0b
content:
  -
    type: paragraph
    content:
      -
        type: text
        text: "React error boundaries allow you to prevent your React application from completely crashing in the event of an error in your code.\_"
  -
    type: paragraph
    content:
      -
        type: text
        text: "You can read more about React's error boundaries on their\_"
      -
        type: text
        marks:
          -
            type: link
            attrs:
              href: 'https://reactjs.org/docs/error-boundaries.html'
              target: null
              rel: null
        text: 'documentation page'
      -
        type: text
        text: .
  -
    type: paragraph
    content:
      -
        type: text
        text: 'The key point to understand when using error boundaries is this snippet from the docs:'
  -
    type: blockquote
    content:
      -
        type: paragraph
        content:
          -
            type: text
            text: 'catch JavaScript errors anywhere in their child component tree'
  -
    type: paragraph
    content:
      -
        type: text
        text: 'Specifically, "component tree". If you try to use Error Boundaries in the following way, none of your errors will be caught:'
  -
    type: set
    attrs:
      values:
        type: code
        code: |
          function SomeComponent() {
             return (
              	<CustomErrorCatcher>
                  	{invalidVariable}
                  </CustomerErrorCatcher>
              );
          }
        syntax: javascript
  -
    type: paragraph
    content:
      -
        type: text
        text: "In the example above,\_"
      -
        type: text
        marks:
          -
            type: code
        text: CustomErrorCatcher
      -
        type: text
        text: "\_is our class component that implements our error boundary logic."
  -
    type: paragraph
    content:
      -
        type: text
        text: "The code that is throwing an error,\_"
      -
        type: text
        marks:
          -
            type: code
        text: '{invalidVariable}'
      -
        type: text
        text: ", is rendering inside the\_"
      -
        type: text
        marks:
          -
            type: code
        text: CustomErrorCatcher
      -
        type: text
        text: "\_component, instead of in a child component. This is what prevents the error boundary from catching the error. The error boundary\_"
      -
        type: text
        marks:
          -
            type: italic
        text: 'will only'
      -
        type: text
        text: "catch errors in child\_"
      -
        type: text
        marks:
          -
            type: bold
        text: components
      -
        type: text
        text: .
  -
    type: paragraph
    content:
      -
        type: text
        text: 'So instead, I recommend you do something like this at the top-level of your application:'
  -
    type: set
    attrs:
      values:
        type: code
        code: |
          function Root() {
              return (
              	<CustomErrorCatcher>
                  	<App />
                  </CustomErrorCatcher>
              );
          }
          ReactDom.render(<Root />, document.getElementById('app'));
        syntax: javascript
  -
    type: paragraph
    content:
      -
        type: text
        text: 'This way your entire application is wrapped in an error boundary, in the event that any of your components throw an error.'
---
